worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -371,17 +371,30 @@ function isEmptyMap(mapValue) {
     return jsonString === '{}';
 }
 
+function searchFieldsForValue(fieldValues, value) {
+    if (fieldValue.isStringValue() && fieldValue.getStringValue() === value) {
+        return true;
+    }
+    return false;
+}
+
 function fieldHasStringValue(document, fieldName, value) {
+    var fieldValues = document.getField(fieldName).getValues();
+    for(fieldValue of fieldValues) {
+        return searchFieldsForValue(fieldValues, value);
+    }
+}
 
+function fieldHasAnyStringValue(document, fieldName, values) {
+    var valueFound = false;
     var fieldValues = document.getField(fieldName).getValues();
-    if (fieldValues)
-    {
-        for (var fieldValue of fieldValues) {
-            if (fieldValue.isStringValue() && fieldValue.getStringValue() === value) {
-                return true;
+
+    for(fieldValue of fieldValues) {
+        for (var value of values) {
+            if(searchFieldsForValue(fieldValues, value)) {
+                valueFound = true;
```

Comment:
You can just exit early from the function at this point (i.e. with `return true;`).  The way you have it at the minute it would keep searching unnecessarily,

Reply:
@dermot-hardy yes good point, could do this instead:


    for(fieldValue of fieldValues) {
        for (var value of values) {
            if(searchFieldsForValue(fieldValues, value)) {
                return true;
            }
        }
    }
    return false;